clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -123,6 +126,34 @@ public static Cluster clusterWith(final int nodes, final String topic, final int
         return clusterWith(nodes, Collections.singletonMap(topic, partitions));
     }
 
+    public static MetadataCache metadataCacheWith(final int nodes, final Map<String, Integer> topicPartitionCounts) {
```

Comment:
Useful to have a brief javadoc to explain replica counts and assignment strategy.

Reply:
done